ac_posystem/src/pages/MainCategory.js

```diff
@@ -3,103 +3,58 @@ import "../assets/css/fontawesome.min.css";
+import image1 from "../assets/images/product-1.png";
+import image2 from "../assets/images/product-2.png";
+import image3 from "../assets/images/product-3.png";
+import image4 from "../assets/images/product-4.png";
+import image5 from "../assets/images/product-5.png";
+import image6 from "../assets/images/product-6.png";
+import image7 from "../assets/images/product-7.png";
+import image8 from "../assets/images/product-8.png";
+import image9 from "../assets/images/product-9.png";
 import axios from "axios";
 import { useState } from "react";
 import { useEffect } from "react";
 import { useLocation } from "react-router-dom";
+
 import { Link } from "react-router-dom";
 
 var cat = "";
 var datalist = [];
 var addTOCart = [];
-
 const MainCategory = () => {
   const location = useLocation();
   cat = location.state.id;
-  console.log("9999 cat_id: ", cat);
-
-  const [counter, setCount] = useState(0);
-
-  let incrementCount = (e) => {
-    setCount(counter + 1);
-    //jquery for showing popup
-    const modal = document.querySelector(".modal");
-    const closeBtn = document.querySelector(".close");
-    modal.style.display = "block";
-    closeBtn.addEventListener("click", () => {
-      modal.style.display = "none";
-    });
-
-    const mapping = { b1: 0, b2: 1, b3: 2, b4: 3, b5: 4, b6: 5 };
-    const a = e.target.id;
-    const index = mapping[a];
-    console.log("INDEX == ", index);
-    var item = addTOCart.find((x) => x.sc_id == datalist[index].sc_id);
-    if (item) {
-      item.count = item.count + 1;
-    } else {
-      var item = JSON.parse(JSON.stringify(datalist[index])); //just to make sure its not passing reference
-      item.count = 1;
-      addTOCart.push(item);
-    }
-  };
-
-  let decrementCount = (e) => {
-    if (counter <= 0) {
-      setCount(0);
-    } else {
-      setCount(counter - 1);
-    }
-    const mapping = { b11: 0, b12: 1, b13: 2, b14: 3, b15: 4, b16: 5 };
-    const a = e.target.id;
-    const index = mapping[a];
-    console.log("INDEX == ", index);
-    var item = addTOCart.find((x) => x.sc_id == datalist[index].sc_id);
-    if (item) {
-      if (item.count == 0) {
-        item.count = 0;
-      } else {
-        item.count = item.count - 1;
-      }
-    } else {
-      var item = JSON.parse(JSON.stringify(datalist[index])); //just to make sure its not passing reference
-      item.count = 1;
-      addTOCart.pop(item);
-    }
-    console.log("INDEX == ", addTOCart);
-  };
+  const catt = JSON.parse(sessionStorage.getItem("cat"));
 
+  if (cat != catt) {
+    cat = catt;
+  }
+  console.log("9999 cat: ", cat);
   useEffect(() => {
     const fetchSubCategories = async () => {
       try {
         const res = await axios.get(
-          "http://localhost:8803/sub_categories/" + location.state.id
+          "http://localhost:8803/sub_categories/" + cat
         );
         datalist = res.data;
         document.getElementById("h1").innerHTML = res.data[0].sc_name;
```

Comment:
Never assign static values to restrict website, instead make it dynamic and assign values in dynamic way.